### ISSUE TITLE
Signed Integer Overflow when handing over large amount to reader

### ISSUE THREAD

**Issue by pc-coholic** [iZettle/sdk-android#4]
# Expected Behavior

Making a payment request for [100.000.000,00](https://i.imgur.com/QBSdxTV.png) Euro should prompt a payment for exactly that amount on the reader.

# Current Behavior

Signed 32bit integer overflow

![IMG_20200616_214135](https://user-images.githubusercontent.com/654548/86166398-02913700-bb15-11ea-86c5-f5748764a080.jpg)

# Steps to Reproduce

````kotlin
val paymentIntent = CardPaymentActivity.IntentBuilder(activity)
                .amount((receipt_total * BigDecimal("100.00")).toLong())  // TODO: Does not work for 0-digit currencies
                .reference(reference)
                .enableTipping(false)
                .build()
````

## Context 

* Operating Version: Android 10
* SDK version: 1.4.0
* Originally reported as API-16526, reposted here to tracking purposes

**Comment by dmilovanovi** [comment]
Hey,

Thank you for reaching out to us.

Our team will have a look and we'll get back to you as soon as possible.


Kind regards.

**Comment by pvoid** [comment]
Fixed in version `1.8.3`
